src/screens/FindFriends/index.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Alert, ListRenderItem, StyleSheet, View } from 'react-native';
+import { FormControl, useTheme, IconButton, useToast } from 'native-base';
+import { useIsFocused } from '@react-navigation/native';
+import { RowMap, SwipeListView } from 'react-native-swipe-list-view';
+import { Controller, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { MagnifyingGlass, XCircle } from 'phosphor-react-native';
+import { FirebaseFirestoreTypes } from '@react-native-firebase/firestore';
+import firestore from '@react-native-firebase/firestore';
+import auth, { FirebaseAuthTypes } from '@react-native-firebase/auth';
+
+import Toast from '@components/Toast';
+import VStack from '@components/VStack';
+import Input from '@components/Input';
+import UserCard from '@components/UserCard';
+import HiddenButton from '@components/HiddenButton';
+import { ProfileDto } from '@interfaces/profile.dto';
+import { useAppDispatch } from '@store/index';
+import { setTitle } from '@store/slices/navigation-slice';
+import {
+  AXIS_X_PADDING_CONTENT,
+  NO_LABEL_INPUT_MARGIN_BOTTOM,
+  TOAST_DURATION,
+  VERTICAL_PADDING_LISTS,
+} from '@utils/constants';
+
+type FormData = {
+  searchValue: string;
+};
+
+const schema = yup.object().shape({
+  searchValue: yup.string(),
+});
+
+const FindFriends = () => {
+  const toast = useToast();
+  const dispatch = useAppDispatch();
+  const isFocused = useIsFocused();
+  const { colors } = useTheme();
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+  } = useForm<FormData>({ resolver: yupResolver(schema) });
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [users, setUsers] = useState<ProfileDto[]>([]);
+  const profilesRef = useRef<
+    FirebaseFirestoreTypes.CollectionReference<ProfileDto>
+  >(firestore().collection<ProfileDto>('profiles'));
+  const userSession: FirebaseAuthTypes.User = auth().currentUser!;
+
+  useEffect(() => {
+    let isMounted = true;
```

Comment:
this shouldn't be a state?

Reply:
isMounted is a variable to avoid memory leak on useEffect